packages/snaps-jest/src/matchers.ts

```diff
@@ -158,33 +158,142 @@ export const toRespondWithError: MatcherFunction<[expected: Json]> = function (
  * is intended to be used with the `expect` global.
  *
  * @param actual - The actual response.
- * @param expected - The expected notification message.
- * @param type - The expected notification type.
+ * @param expectedMessage - The expected notification message.
+ * @param expectedType - The expected notification type.
+ * @param expectedTitle - The expected notification title.
+ * @param expectedContent - The expected notification JSX content.
+ * @param expectedFooterLink - The expected footer link object.
  * @returns The status and message.
  */
 export const toSendNotification: MatcherFunction<
-  [expected: string, type?: EnumToUnion<NotificationType> | undefined]
-> = function (actual, expected, type) {
+  [
+    expectedMessage: string,
+    expectedType: EnumToUnion<NotificationType>,
+    expectedTitle?: string | undefined,
+    expectedContent?: JSXElement | undefined,
+    expectedFooterLink?: { text: string; href: string } | undefined,
+  ]
+> = function (
+  actual,
+  expectedMessage,
+  expectedType,
+  expectedTitle,
+  expectedContent,
+  expectedFooterLink,
+) {
   assertActualIsSnapResponse(actual, 'toSendNotification');
 
   const { notifications } = actual;
-  const pass = notifications.some(
-    (notification) =>
-      this.equals(notification.message, expected) &&
-      (type === undefined || notification.type === type),
-  );
+  let jsxContent: JSXElement | undefined;
 
-  const message = pass
-    ? () =>
-        `${this.utils.matcherHint('.not.toSendNotification')}\n\n` +
-        `Expected: ${this.utils.printExpected(expected)}\n` +
-        `Expected type: ${this.utils.printExpected(type)}\n` +
-        `Received: ${this.utils.printReceived(notifications)}`
-    : () =>
-        `${this.utils.matcherHint('.toSendNotification')}\n\n` +
-        `Expected: ${this.utils.printExpected(expected)}\n` +
-        `Expected type: ${this.utils.printExpected(type)}\n` +
-        `Received: ${this.utils.printReceived(notifications)}`;
+  if (hasProperty(actual, 'getInterface')) {
+    jsxContent = (actual.getInterface as () => SnapInterface)().content;
```

Comment:
Is this type cast necessary?

Reply:
Hmm this is another example of the type narrowing not working with `hasProperty`, it reads it as `unknown`. I just changed this to use the `in` operator and I don't need to cast.